storybook/stories/playground/zoom-out/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { useDispatch } from '@wordpress/data';
+import {
+	BlockEditorProvider,
+	BlockCanvas,
+	store as blockEditorStore,
+	BlockList,
+} from '@wordpress/block-editor';
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/private-apis';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import editorStyles from '../editor-styles';
+// eslint-disable-next-line @wordpress/dependency-group
+import contentCss from '!!raw-loader!../../../../packages/block-editor/build-style/content.css';
+import { pattern } from './pattern';
+
+// Temporary hack to access private APIs before stabilizing zoom level.
+const { unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
+	'@wordpress/edit-site'
+);
+
+function EnableZoomOut() {
+	const { setZoomLevel } = unlock( useDispatch( blockEditorStore ) );
+
+	useEffect( () => {
+		setZoomLevel( 50 );
+	}, [ setZoomLevel ] );
+
+	return null;
+}
+
+export default function EditorZoomOut() {
+	const [ blocks, updateBlocks ] = useState( [] );
+
+	useEffect( () => {
+		registerCoreBlocks();
+		updateBlocks( parse( pattern ) );
+	}, [] );
+
+	return (
+		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
+		<div
+			className="editor-zoom-out"
+			onKeyDown={ ( event ) => event.stopPropagation() }
+			style={ { background: '#ddd', border: '1px solid gray' } }
+		>
+			<BlockEditorProvider
+				value={ blocks }
+				onInput={ updateBlocks }
+				onChange={ updateBlocks }
+			>
+				<EnableZoomOut />
+				<BlockCanvas height="500px" styles={ editorStyles }>
+					<style>{ contentCss }</style>
```

Comment:
This highlighted that third-party block editors need to load the "content.css" file from the block-editor package within the iframe. I would love if this was not necessary (and absorbed by the component automatically instead). cc @ellatrix in case you have ideas.

Reply:
How is this related to this [config](https://github.com/WordPress/gutenberg/blob/trunk/storybook/package-styles/config.js)? I'm still a bit confused about how to properly inject styles to stories. I guess the config is about adding styles to the main `storybook` iframe, right? --cc @ciampo @mirka .

Reply:
Not sure I'm following. Shouldn't it be merged into `editorStyles`?

Reply:
@ellatrix No, `editorStyles` seem to be rewritten to add the wrapper... and I think that breaks some of the "content" styles that we provide. 

@ntsekouras not related to these styles at all, because the styles here are the "content" styles that should apply within the iframe, the rest is just editor styles, ui styles...

Reply:
@youknowriad What does that break? Ideally, everything should receive the content wrapper. For whatever styles that it is not the case now it will have to be redone at some point. It shouldn't break the styles -- it should merely add the prefix that adds zero specificity.

Reply:
`.block-editor-iframe__html.is-zoomed-out` this style targets the HTML element of the iframe, so adding the wrapper breaks it. 

The other thing here is that for these styles that are necessary to make the "block-editor" package work, it shouldn't be the responsibility of the implementor to include them manually, ideally, when using the BlockEditor packages, these styles should be there by default without requiring us to inject a CSS file or something. So I'm wondering if we should actually just build these styles as a `string` and auto-inject them in the iframe, rather than use our regular style building pipeline.

I'm also curious how this "content" styles get loaded in the context of the WP block editors.

Reply:
@youknowriad So there's currently an unstable setting `__unstableResolvedAssets` where you can provide scripts and styles. But ideally in the future, everything should go through the normal styles array (with maybe an option to provide a URL instead of inline CSS). And I guess a similar array for scripts. One of the issues is actually that the way we load `__unstableResolvedAssets` is better from a UX and performance perspective. They are provided as as HTML tags (1) because that's the only thing WP can give us after resolving dependencies and (2) because we anyway need them as HTML to prefill the iframe contents (outside of React). At some earlier point we were actually parsing the HTML and injecting it all in React, but that caused unsettled content glitches and was less performant than what we do now. Maybe I should create an issue to discuss this further.

Reply:
My thinking is that `__unstableResolvedAssets` shouldn't be needed at all. We should either have:

 - Styles that are provided by the framework should be loaded without including them in any setting. (The ones I loaded in this PR fall into this category)
 - The rest should just load in the regular `styles` property, it's fine if no wrapper is added... that's an implementation detail.

Reply:
Oh sure, I see what you mean. Yes it should work automatically.

Reply:
We currently rely on the server side dependency resolver to provide these. So you're saying that at the very least the block editor should automatically add it's own stylesheet and it's dependencies right?

Reply:
In that case, we'll need (some of) these styles:

https://github.com/WordPress/wordpress-develop/blob/1508e5bf0fa3ad2d001de08eab4dee49985bb322/src/wp-includes/script-loader.php#L1682

How could we load the components styles for example?

One idea would be to also remove the need for these components styles. I believe it's only needed for things like the placeholder, but this should be rendered in shadow DOM instead of the iframe.

Reply:
Good list:

Indeed, there's a few things on that list that are weird:

 - components make sense for placeholders
 - why are we loading "editor", "reusable blocks" and "patterns"?
 - block library seems more like "third-party" styles that should be loaded through the "styles" property.

Reply:
* That's weird, I remember removing `wp-editor` and here it is indeed: #33496. It probably wasn't backported. I'll create a backport PR.
* Looking at reusable blocks, this only contains a single rule that is for a modal (so nothing in the content). I'll remove it, and also backport it (in the same PR since we're in RC anyway).
* Patterns also seems to be for modals and a button. Could be that that button is part of the placeholder, I'll check.

Generally for UI, it would simplify a lot if we used shadow DOM so we don't need components and UI styling at all.

Reply:
Here's the one for reusable blocks: #66285.

Reply:
The backport for `wp-editor` removal: https://github.com/WordPress/wordpress-develop/pull/7609

Reply:
Here's the removal of the patterns (UI) stylesheet: #66306. It's indeed all UI related.

Reply:
One the questions is still: how would we load the components (for now) and block library styles directly from inside the `block-editor` package? Do we "raw" import the CSS or something?